googlefirestore/src/main/java/com/yahoo/ycsb/db/firestore/GoogleFirestoreClient.java

```diff
@@ -0,0 +1,246 @@
+    DocumentReference docRef = toReference(table, key);
+
+    try {
+      DocumentSnapshot docSs = docRef.get().get();
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("read: Collection: " + table + " document: " + key);
+      }
+      if (docSs.exists()) {
+        parseFields(fields, docSs, result);
+        return Status.OK;
+      } else {
+        LOGGER.error("read: Referenced document is missing. : " + docSs.getReference().toString());
+        return Status.ERROR;
+      }
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      LOGGER.error("Interrupted during read().", e);
+      return Status.ERROR;
+    } catch (ExecutionException e) {
+      LOGGER.error("Error during read().", e);
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status scan(String table, String startkey, int recordcount, Set<String> fields,
+      Vector<HashMap<String, ByteIterator>> result) {
+    Query query = fsDb.collection(table)
+        .orderBy("__name__")
+        .startAt(startkey)
+        .limit(recordcount);
+
+    try {
+      QuerySnapshot querySs = query.get().get();
+      for (DocumentSnapshot docSs : querySs.getDocuments()) {
+        HashMap<String, ByteIterator> scanres = new HashMap<>();
+        parseFields(fields, docSs, scanres);
+        result.add(scanres);
+      }
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("scan: " + startkey + " : " + recordcount);
+      }
+      return Status.OK;
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      LOGGER.error("Interrupted during scan().", e);
+      return Status.ERROR;
+    } catch (ExecutionException e) {
+      LOGGER.error("Error during scan().", e);
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status update(String table, String key, Map<String, ByteIterator> values) {
+    DocumentReference docRef = toReference(table, key);
+    Map<String, Object> data = toData(values);
+
+    try {
+      docRef.set(data, SetOptions.merge()).get();
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("update: Document: " + key + " : " + data.toString());
+      }
+      return Status.OK;
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      LOGGER.error("Interrupted during update().", e);
+      return Status.ERROR;
+    } catch (ExecutionException e) {
+      LOGGER.error("Error during update().", e);
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status insert(String table, String key, Map<String, ByteIterator> values) {
+    DocumentReference docRef = toReference(table, key);
+    Map<String, Object> data = toData(values);
+
+    try {
+      ApiFuture<WriteResult> writeResult = docRef.set(data, SetOptions.merge());
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("insert: Update time: " + writeResult.get().getUpdateTime());
```

Comment:
if debug isn't enabled for the logger, don't we still need to call `writeResult.get()` ?

Reply:
Made it do the get by default to be more consistent with the rest of the code.